lib/core/data/event/repository/event_repository_impl.dart

```diff
@@ -587,4 +596,104 @@ class EventRepositoryImpl implements EventRepository {
       ),
     );
   }
+
+  @override
+  Future<Either<Failure, List<EventRole>>> getEventRoles() async {
+    final result = await client.query$GetEventRoles(
+      Options$Query$GetEventRoles(
+        fetchPolicy: FetchPolicy.noCache,
+      ),
+    );
+    if (result.hasException) {
+      return Left(Failure.withGqlException(result.exception));
+    }
+    return Right(
+      (result.parsedData?.getRoles ?? [])
+          .map((item) => EventRole.fromJson(item.toJson()))
+          .toList(),
+    );
+  }
+
+  @override
+  Future<Either<Failure, List<EventUserRole>>> getListUserRole({
+    required String eventId,
+    String? roleId,
+    String? searchCriteria,
+  }) async {
+    final result = await client.query$GetListUserRole(
+      Options$Query$GetListUserRole(
+        fetchPolicy: FetchPolicy.noCache,
+        variables: Variables$Query$GetListUserRole(
+          input: Input$EventRoleFilter(
+            event_id: eventId,
+          ),
+        ),
+      ),
+    );
+    if (result.hasException) {
+      return Left(Failure.withGqlException(result.exception));
+    }
+    return Right(
+      List.from(
+        result.parsedData!.getListUserRole
```

Comment:
be careful when using !,  there will be case where we dont have exception, but parsedData is null